tools/shoestring/shoestring/wizard/setup_file_generator.py

```diff
@@ -129,3 +136,100 @@ def patch_shoestring_config(shoestring_filepath, package_config_filepath):
+	bootstrap = screens.get('bootstrap')
+
+	if 'dual' != node_type:
+		return False
+
+	with open(Path(bootstrap.path) / 'gateways/rest-gateway/rest.json', 'rt', encoding='utf8') as file:
+		data = json.load(file)
+		if 'nodeMetadata' in data:
+			with open(output_filename, 'wt', encoding='utf8') as outfile:
+				json.dump({'nodeMetadata': data['nodeMetadata']}, outfile, indent=2)
+
+			return True
+
+	return False
+
+
+def _get_resources_configuration_value(bootstrap_path, extension, search_identifiers):
+	bootstrap_configuration_manager = ConfigurationManager(Path(bootstrap_path) / 'nodes/node/server-config/resources')
+	return bootstrap_configuration_manager.lookup(f'config-{extension}.properties', search_identifiers)
+
+
+def prepare_overrides_file_from_bootstrap(screens, output_filename):  # pylint: disable=invalid-name
+	"""Prepares an override file based on current settings."""
+
+	bootstrap = screens.get('bootstrap')
+	values = _get_resources_configuration_value(bootstrap.path, 'node', [
+		('node', 'minFeeMultiplier'),
+		('localnode', 'host'),
+		('localnode', 'friendlyName')
+	])
+	with open(output_filename, 'wt', encoding='utf8') as outfile:
+		outfile.write('\n'.join([
+			'[node.node]',
+			f'minFeeMultiplier = {values[0]}',
+			'',
+			'[node.localnode]',
+			f'host = {values[1]}',
+			f'friendlyName = {values[2]}',
+		]))
+
+
+async def prepare_shoestring_files_from_bootstrap(screens, shoestring_directory):  # pylint: disable=invalid-name
+	"""Prepares shoestring configuration files based on bootstrap settings."""
+
+	network_type = screens.get('network-type').current_value
+	node_type = screens.get('node-type').current_value
+	bootstrap = screens.get('bootstrap')
+
+	config_filepath = shoestring_directory / 'shoestring.ini'
+	await prepare_shoestring_config(network_type, config_filepath)
+
+	node_features = NodeFeatures.PEER
+	if node_type in ['dual', 'light']:
+		node_features |= NodeFeatures.API
+	await patch_bootstrap_shoestring_config(bootstrap.path, config_filepath, bootstrap.include_node_key)
+
+	resources_path = Path(bootstrap.path) / 'nodes/node/server-config/resources'
+
+	def is_feature_enabled(extension, config_property):
+		bootstrap_configuration_manager = ConfigurationManager(resources_path)
+		return 'true' == bootstrap_configuration_manager.lookup(f'config-{extension}.properties', [config_property])[0]
+
+	if is_feature_enabled('harvesting', ('harvesting', 'enableAutoHarvesting')):
+		node_features |= NodeFeatures.HARVESTER
+
+	if is_feature_enabled('finalization', ('finalization', 'enableVoting')):
+		node_features |= NodeFeatures.VOTER
+
+	values = _get_resources_configuration_value(bootstrap.path, 'node', [
+		('localnode', 'host'),
+		('localnode', 'friendlyName')
+	])
+	if not is_hostname(values[0]) and not is_ip_address(values[0]):
+		raise ValueError(f'Shoestring requires valid IP address or domain name. host: {values[0]}.')
+
+	api_https = not is_ip_address(values[0]) and is_hostname(values[0])
```

Comment:
does bootstrap only support hostname with https?

Reply:
to create a cert you need a valid hostname and domain.

The problem with bootstrap is that the hostname/ip is not required so it might be empty.  I was thinking to show the node-setting so the user will always to enter this info? 🤷🏾‍♂️

Reply:
i was more referring to check in `_resolve_hostname_and_configure_https` where it allows hostname + http. here, it looks like https is used whenever hostname is provided.

not sure if something we need to change, but just pointing out.